op-e2e/e2eutils/waits.go

```diff
@@ -78,3 +78,23 @@ func WaitFor(ctx context.Context, rate time.Duration, cb func() (bool, error)) e
 		}
 	}
 }
+
+func WaitAndGet[T interface{}](ctx context.Context, pollRate time.Duration, get func() (T, error), predicate func(T) bool) (*T, error) {
```

Comment:
Then you need some way to create a `T` which generics don't let you do. ie you can't do `return nil, err` anymore. This seemed like the simplest solution rather than passing in a factory to create nil T values.

Reply:
I think I've done it by doing `var ret T` and then returning T

Reply:
https://go.dev/play/p/YXeLmVN1TOq?v=gotip

Reply:
ah nice. :) Done.